src/uwtools/config/formats/ini.py

```diff
@@ -62,6 +63,8 @@ def dump_dict(path: OptionalPath, cfg: dict) -> None:
         # when an INI contains multiple sections. Unfortunately, it also adds a newline after the
         # _final_ section, resulting in an anomalous trailing newline. To avoid this, write first to
         # memory, then strip the trailing newline.
+        target_format = get_file_type(str(path))
```

Comment:
I suspect trouble if `path` is `None`, which one of the acceptable values of `OptionalPath`. In that case, we'd be streaming the output to `stdout`, but would still want to do a depth check. I think the typechecker would have complained about this without the `str()` wrapper, as the type of the input argument of `get_file_type()` is `DefinitePath`, which rules out a `None` value.

But do we even need to call `get_file_type()`? This class is `INIConfig`, so don't we _know_ the type?

Reply:
Hmm unfortunately the input to `get_file_type()` is `DefinitePath` - I had a fixed type originally, then made it more complicated here, then kind of decided that I liked the soft requirement of a filetype suffix that this adds. Could hard-code it here but check for a suffix in `dump`?

Reply:
Combined with my other suggestion about breaking `config_check_depths()` up into mode-specific functions, could these two lines become just
```
config_check_depths_dump(config_obj=cfg, target_format=FORMAT.ini)
```
?

Reply:
Yep, looks nice and clean now!